nano/node/confirming_set.cpp

```diff
@@ -133,10 +133,14 @@ void nano::confirming_set::run_batch (std::unique_lock<std::mutex> & lock)
 		notification.already_cemented.swap (already);
 
 		// Wait for the worker thread if too many notifications are queued
-		while (notification_workers.num_queued_tasks () >= config.max_queued_notifications)
+		std::unique_lock lock{ mutex };
+		bool stopped_l;
+		condition.wait (lock, [this, &stopped_l] {
+			return (stopped_l = stopped) || notification_workers.num_queued_tasks () < config.max_queued_notifications;
```

Comment:
If this is using condition variables now, then condition needs to be notified after each processed entry

Reply:
What do you think of using the wait_for variant?

Reply:
I think the impact of additional notifications will be zero, this is bound by IO anyway.

Reply:
thread_pool doesn't have a notification when the number of queue tasks changes so I'll leave this to poll using wait_for instead of sleep.

Reply:
Is this the same? Won't `condition.wait_for (lock, 100ms,` wait for 100ms and then continue regardless of whether the predicate is fulfilled? 

It's true that there is no event for thread pool, but that notification can be triggered here:
```
 notification_workers.push_task ([this, notification = std::move (notification)] () {
			stats.inc (nano::stat::type::confirming_set, nano::stat::detail::notify);
			batch_cemented.notify (notification);
                        condition.notify ();
		});
```

Reply:
In your code the `stats.inc (nano::stat::type::confirming_set, nano::stat::detail::cooldown);` is lost, which is necessary to properly understand live node bottlenecks. Code below should be most similar to the previous logic:

```
while (notification_workers.num_queued_tasks () >= config.max_queued_notifications)
 {
 	stats.inc (nano::stat::type::confirming_set, nano::stat::detail::cooldown);
 	condition.wait_for(..., 100ms, [...] { return stopped; }
 	if (stopped) { return; }
 }
 ```